Sources/ComposableNavigation/ModalNavigation/ModalNavigationHandler.swift

```diff
@@ -104,23 +108,30 @@ public class ModalNavigationHandler<ViewProvider: ViewProviding>: NSObject, UIAd
 		return viewController
 	}
 	
-	private func dimissModal(from presentingViewController: UIViewController) {
+	private func dismissModal(
+		from presentingViewController: UIViewController,
+		animated: Bool
+	) {
 		// Prevent dismissal of unrelated view controller
 		if presentingViewController.presentedViewController == currentViewControllerItem?.viewController {
-			presentingViewController.dismiss(animated: shouldAnimateModalChanges, completion: nil)
+			presentingViewController.dismiss(animated: animated, completion: nil)
 		}
 		currentViewControllerItem = nil
 	}
 	
-	private var shouldAnimateModalChanges: Bool {
-		UIView.areAnimationsEnabled
+	private func shouldAnimateChanges(state: ModalItemNavigation.State) -> Bool {
+		if !UIView.areAnimationsEnabled {
+			return false
+		} else {
```

Comment:
I wonder if it is necessary to check whether the global switch for animations is switched off, because if it is, then the 'present(viewController:animated:) will not be animated anyway. So, we kind of double killing animations by doing it this way, but perhaps I'm missing a detail.

Reply:
I had the same feeling but at least the UINavigationController is behaving differently as the unit tests start falling when I removed that check. So I kept it for consistency's sake for the other NavigationHandlers as well.